src/wit_gen.rs

```diff
@@ -42,15 +42,25 @@ pub struct TraceWitness<F> {
 #[derive(Debug, Default)]
 pub struct TraceContext<F> {
     witness: TraceWitness<F>,
+    num_steps: usize,
 }
 
 impl<F> TraceContext<F> {
+    pub fn new(num_steps: usize) -> Self {
+        Self {
+            witness: TraceWitness {
+                step_instances: Vec::new(),
+            },
+            num_steps,
+        }
+    }
+
     pub fn get_witness(self) -> TraceWitness<F> {
         self.witness
     }
 }
 
-impl<F> TraceContext<F> {
+impl<F: Clone> TraceContext<F> {
```

Comment:
Is f: clone required anymore?

Reply:
no, that was for a feature I tried and forgot to remove.